pkg/registry/batch/job/strategy.go

```diff
@@ -115,8 +114,8 @@ func (jobStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object
 		newJob.Spec.TTLSecondsAfterFinished = nil
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.IndexedJob) && oldJob.Spec.CompletionMode == batch.NonIndexedCompletion {
-		newJob.Spec.CompletionMode = batch.NonIndexedCompletion
+	if !utilfeature.DefaultFeatureGate.Enabled(features.IndexedJob) && oldJob.Spec.CompletionMode == nil {
+		newJob.Spec.CompletionMode = nil
```

Comment:
this would allow updating from NonIndexed->Indexed even when the feature gate is disabled, no?

Reply:
It would... but field is immutable (checked in `ValidateJobSpecUpdate`)

Also, the feature gate is, in a way, protecting the field to not be set, without caring about the value.